internal/services/dns/dns_zone_resource.go

```diff
@@ -362,3 +376,17 @@ func flattenArmDNSZoneSOARecord(input *recordsets.RecordSet) []interface{} {
 
 	return output
 }
+
+func soaRecordHostName(ctx context.Context, timeout time.Duration, recordSetsClient *recordsets.RecordSetsClient, id zones.DnsZoneId) (*string, error) {
+	soaRecord := recordsets.NewRecordTypeID(id.SubscriptionId, id.ResourceGroupName, id.DnsZoneName, recordsets.RecordTypeSOA, "@")
+	soaRecordResp, err := recordSetsClient.Get(ctx, soaRecord)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	if soaRecordResp.Model.Properties.SOARecord.Host == nil {
```

Comment:
Can we also check any possible nilable properties along the chain (e.g. the `SOARecord`)?

Reply:
i have done it all along the chain now, thanks